ext/geos_c_impl/geometry.c

```diff
@@ -759,8 +759,8 @@ method_geometry_buffer_with_style(VALUE self,
                             self_geom,
                             rb_num2dbl(distance),
                             RGEO_FACTORY_DATA_PTR(factory)->buffer_resolution,
-                            rb_num2int(endCapStyle),
-                            rb_num2int(joinStyle),
+                            RB_NUM2INT(endCapStyle),
```

Comment:
The `rb_num2int` function actually returns a long, where the `RB_NUM2INT` macro try to return an int, and cast the long into int if failed